sbc_compassion/models/import_letters_history.py

```diff
@@ -233,57 +155,62 @@ def onchange_config(self):
-        - check for duplicate file names and skip them
-        - decompress zip file if necessary
-        - call _analyze_attachment for every resulting file
-        """
+    def run_analyze(self, generator=None):
+        if generator is None:
+            generator = self.local_generator
+
         self.ensure_one()
-        # keep track of file names to detect duplicates
-        file_name_history = []
-        logger.info("Imported files analysis started...")
-        progress = 1
+        self.state = "pending"
+        logger.info("Letters import started...")
 
-        for attachment in self.data:
-            if attachment.name not in file_name_history:
-                file_name_history.append(attachment.name)
-                file_data = base64.b64decode(
-                    attachment.with_context(bin_size=False).datas
-                )
-                # check for zip
-                if func.check_file(attachment.name) == 2:
-                    zip_file = BytesIO(file_data)
-                    zip_ = zipfile.ZipFile(zip_file, "r")
-                    for f in zip_.namelist():
-                        logger.debug(f"Analyzing file {progress}/{self.nber_letters}")
-                        self._analyze_attachment(zip_.read(f), f)
-                        progress += 1
-                # case with normal format (PDF,TIFF)
-                elif func.check_file(attachment.name) == 1:
-                    logger.debug(f"Analyzing file {progress}/{self.nber_letters}")
-                    self._analyze_attachment(file_data, attachment.name)
-                    progress += 1
-                else:
-                    raise UserError(_("Only zip/pdf files are supported."))
-            else:
-                raise UserError(_("Two files are the same"))
+        for i, n, file in generator():
+            logger.info(f"{i}/{n} : {file}")
 
+        logger.info(f"Letters import completed !")
         # remove all the files (now they are inside import_line_ids)
         self.data.unlink()
         self.import_completed = True
-        logger.info("Imported files analysis completed.")
 
-    def _analyze_attachment(self, file_data, file_name):
-        line_vals = func.analyze_attachment(
-            self.env, file_data, file_name, self.template_id
-        )
-        for i in range(0, len(line_vals)):
-            line_vals[i]["import_id"] = self.id
-            self.env["import.letter.line"].create(line_vals[i])
+    def _analyze_pdf(self, pdf_data, file_name):
+        try:
+            partner_code, child_code, preview = read_barcode.letter_barcode_detection_pipeline(pdf_data)
+        except Exception as e:
+            logger.error(f"Couldn't import file {file_name} : \n{traceback.format_exc()}")
+            return
+
+        partner = self.env["res.partner"].search([("ref", "=", partner_code)], limit=1)
+        if partner:
+            partner_id = partner.id
+        else:
+            partner_id = None
```

Comment:
This doesn't seem required. If no partner is found, `partner.id` will return False.